src/sentry/api/serializers/models/user.py

```diff
@@ -143,23 +141,30 @@ class DetailedUserSerializer(UserSerializer):
     def get_attrs(self, item_list, user):
         attrs = super(DetailedUserSerializer, self).get_attrs(item_list, user)
 
-        authenticators = defaultdict(list)
-        queryset = Authenticator.objects.filter(
+        # ignore things that aren't user controlled (like recovery codes)
+        authenticators = manytoone_to_dict(Authenticator.objects.filter(
+            user__in=item_list,
+        ), 'user_id', lambda x: not x.interface.is_backup_interface)
+
+        permissions = manytoone_to_dict(UserPermission.objects.filter(
             user__in=item_list,
-        )
-        for auth in queryset:
-            # ignore things that aren't user controlled (like recovery codes)
-            if auth.interface.is_backup_interface:
-                continue
-            authenticators[auth.user_id].append(auth)
+        ), 'user_id')
 
         for item in item_list:
             attrs[item]['authenticators'] = authenticators[item.id]
+            attrs[item]['permissions'] = permissions[item.id]
 
         return attrs
 
     def serialize(self, obj, attrs, user):
         d = super(DetailedUserSerializer, self).serialize(obj, attrs, user)
+        # XXX(dcramer): we dont use is_active_superuser here as we simply
+        # want to tell the UI that we're an authenticated superuser, and
+        # for requests that require an *active* session, they should prompt
+        # on-demand. This ensures things like links to the Sentry admin can
+        # still easily be rendered.
+        d['isSuperuser'] = user.is_superuser
```

Comment:
*this* is a change in behavior in that before this you could never identify if a user in Sentry was a superuser unless that user was you. I think its an acceptable risk change.